flutter_vlc_player/lib/src/vlc_player_controller.dart

```diff
@@ -114,11 +120,19 @@ class VlcPlayerController extends ValueNotifier<VlcPlayerValue> {
   bool get isReadyToInitialize => _isReadyToInitialize;
   bool _isReadyToInitialize;
 
+  void setIsReadyToInitalize(boolValue) {
+    _isReadyToInitialize = boolValue;
+  }
+
   /// This is just exposed for testing. It shouldn't be used by anyone depending
   /// on the plugin.
   @visibleForTesting
   int get viewId => _viewId;
 
+  void setViewId(viewId) {
+    _viewId = viewId;
+  }
+
```

Comment:
The two added methods break encapsulation - severely. 

They are only used in the `onPlatformViewCreated` listener in the video player Widget.
At the very least we could make that listener a part of this class. This way it would be a bit more clear that it's a listener, and not a user-facing method.

On a second thought, this `onPlatformViewCreated` callback can be used by someone who would want to make a custom widget with the controller. We could leave it public, and add a doc comment explaining how it can be used.

Reply:
fixed